packages/semcom-sdk/lib/component/services/register-component.service.spec.ts

```diff
@@ -96,4 +113,116 @@ describe('RegisterComponentService', () => {
 
   });
 
+  it('should generate a correct tag', async () => {
+
+    mockEvalAndDefine();
+
+    const mockComponent = {
+      uri: 'https://www.google.com/test.js',
+      tag: 'component',
+    } as ComponentMetadata;
+
+    const tag = await service.register(mockComponent);
+    expect(tag.startsWith('semcom-component-')).toBe(true);
+
+  });
+
+  it('should eval and define a component', async () => {
+
+    mockEvalAndDefine();
+
+    const mockComponent = {
+      uri: 'https://www.google.com/test.js',
+      tag: 'component',
+    } as ComponentMetadata;
+
+    await service.register(mockComponent);
+    expect(global.eval).toBeCalledTimes(1);
+    expect(service.isRegistered(mockComponent)).toBeTruthy();
+
+  });
+
+  it('should not reimport an existing component if it was already defined', async () => {
+
+    mockEvalAndDefine();
+
+    const mockComponent = {
+      uri: 'https://www.google.com/test.js',
+      tag: 'component',
+    } as ComponentMetadata;
+
+    await service.register(mockComponent);
+    await service.register(mockComponent);
+
+    expect(global.eval).toBeCalledTimes(1);
+    expect(customElements.define).toBeCalledTimes(1);
+
+  });
+
+  it('should only define a tag once, if two are imported at the same time', async () => {
+
+    mockEvalAndDefine();
+
+    const mockComponent = {
+      uri: 'https://www.google.com/test.js',
+      tag: 'component',
+    } as ComponentMetadata;
+
+    await Promise.all([ service.register(mockComponent), service.register(mockComponent) ]);
+    expect(customElements.define).toBeCalledTimes(1);
+
+  });
+
+  it('should return different html tags for components with different tags', async () => {
+
+    mockEvalAndDefine();
+
+    const mockComponent = {
+      uri: 'https://www.google.com/test.js',
+      tag: 'component',
+    } as ComponentMetadata;
+
+    const mockComponent2 = {
```

Comment:
same comment as the first mockComponent. It's used in multiple tests, so make it a const at the the top of the file.